### PR TITLE
Move admin branding to a distinct shape

### PR DESCRIPTION
**Pull request by TFleury**
Just to allow admin re-branding without overriding whole Layout.

### CODE REVIEW

File: src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml [hunk 1]
```diff
@@ -32,8 +31,7 @@
     <div class="ta-wrapper">
         <div class="ta-navbar-top navbar-expand-lg bg-primary">
             <div class="d-flex w-100">
-                <a class="ta-navbar-brand" href="@Url.Content("~/" + AdminOptions.Value.AdminUrlPrefix)"><img src=@Url.Content("~/TheAdmin/logo.png") alt="@Site.SiteName" /><span>@Site.SiteName</span></a>
-
+                <shape type="Branding" />
```
Comment (deanmarcussen) [inline src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml:34]:
```suggestion
                @await DisplayAsync(await New.Branding())
```

**Review by deanmarcussen — COMMENTED** [review]
@TFleury looks good. can you see this issue https://github.com/OrchardCMS/OrchardCore/issues/8416

and do the same with the ico.

I would also suggest putting the shape in the Admin module, rather than the theme. 
It makes it more overridable, i.e. can be done by another module, rather than having to have an admin theme (obviously can be done through admin templates as well)

**Comment by TFleury** [comment]
> I would also suggest putting the shape in the Admin module, rather than the theme.
It makes it more overridable, i.e. can be done by another module, rather than having to have an admin theme (obviously can be done through admin templates as well)

Not sure we want to override this shape from any module, and it could conflict with front-end shape, or maybe we should rename it to 'AdminBranding'.
AdminTemplates and admin theme override seems to me good enough options to customize branding.

> do the same with the ico

It seems more complicated, it's not possible to register a resource (a link resource in this case) from a shape directly rendered in layout. We need to add the shape in a Zone to make it work.

Another way could be to have an "AdminWidgets" feature, or just to have admin layers. What do you think about this option ?

**Comment by TFleury** [comment]
I found a workaround to allow registration of the icon link, but it's quite dirty :

At the top of the layout :
```razor
@{
    var adminSettings = Site.As<AdminSettings>();
    var darkMode = await DarkModeService.IsDarkModeAsync();
    var brandingHtml = await DisplayAsync(await New.Branding());
}
```
And then :
```razor
<div class="ta-navbar-top navbar-expand-lg bg-primary">
            <div class="d-flex w-100">
                @brandingHtml
                <div class="flex-grow-1">
                    @if (adminSettings.DisplayTitlesInTopbar)
                    {
                        @await RenderSectionAsync("Title", required: false)
                    }
                </div>
```

**Comment by deanmarcussen** [comment]
> rename it to 'AdminBranding'.

sounds good

> At the top of the layout :

also looks good, doesn't feel dirty to me, just how these things work, i.e. call it before the tag helper to render resources.

> Another way could be to have an "AdminWidgets" feature, or just to have admin layers. 

There's a branch here where I made admin layers https://github.com/OrchardCMS/OrchardCore/tree/deanmarcussen/adminlayers. 

I'd literally forgotten about it, so thank you for reminding me, maybe deserves a demo. 

The issue that I ran into (I think) was that I didn't like was that the standard themes widget templates were not discovered, because of course it was in the admin, which started making it seem a bit weird.

 (and then I forgot I wrote it!)

**Comment by agriffard** [comment]
As a reminder, I also tried to provide some admin settings for this purpose but it has been decided not to merge it because we could do ite with Custom Admin Theme and Custom Theme Settings: #5920

File: src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 @inject DarkModeService DarkModeService
-@inject IOptions<AdminOptions> AdminOptions
 @{
     var adminSettings = Site.As<AdminSettings>();
     var darkMode = await DarkModeService.IsDarkModeAsync();
+
+    // Branding is pre-rendred to allow resource injection
```
Comment (deanmarcussen) [inline src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml:6]:
```suggestion
    // Branding is pre-rendered to allow resource injection
```

**Review by deanmarcussen — COMMENTED** [review]
We looked at yesterdays meeting, and it all made sense.

Any chance of some docs, saying the shape is available to override?

**Review by deanmarcussen — APPROVED** [review]

**Pull request merged**